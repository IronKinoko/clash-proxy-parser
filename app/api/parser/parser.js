module.exports = (config) => {
  config['proxy-groups'] = []
  config['rule-providers'] = {
    reject: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/reject.txt',
      path: './ruleset/reject.yaml',
      interval: 86400,
    },
    icloud: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/icloud.txt',
      path: './ruleset/icloud.yaml',
      interval: 86400,
    },
    apple: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/apple.txt',
      path: './ruleset/apple.yaml',
      interval: 86400,
    },
    google: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/google.txt',
      path: './ruleset/google.yaml',
      interval: 86400,
    },
    proxy: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/proxy.txt',
      path: './ruleset/proxy.yaml',
      interval: 86400,
    },
    direct: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/direct.txt',
      path: './ruleset/direct.yaml',
      interval: 86400,
    },
    private: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/private.txt',
      path: './ruleset/private.yaml',
      interval: 86400,
    },
    gfw: {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/gfw.txt',
      path: './ruleset/gfw.yaml',
      interval: 86400,
    },
    'tld-not-cn': {
      type: 'http',
      behavior: 'domain',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/tld-not-cn.txt',
      path: './ruleset/tld-not-cn.yaml',
      interval: 86400,
    },
    telegramcidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/telegramcidr.txt',
      path: './ruleset/telegramcidr.yaml',
      interval: 86400,
    },
    cncidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/cncidr.txt',
      path: './ruleset/cncidr.yaml',
      interval: 86400,
    },
    lancidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt',
      path: './ruleset/lancidr.yaml',
      interval: 86400,
    },
    applications: {
      type: 'http',
      behavior: 'classical',
      url: 'https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/applications.txt',
      path: './ruleset/applications.yaml',
      interval: 86400,
    },
    fanqie: {
      type: 'http',
      behavior: 'classical',
      url: 'https://cdn.jsdelivr.net/gh/ironkinoko/clash-proxy-parser/public/rules/fanqie.yaml',
      path: './ruleset/fanqie.yaml',
      interval: 86400,
    },
    copymanga: {
      type: 'http',
      behavior: 'classical',
      url: 'https://cdn.jsdelivr.net/gh/ironkinoko/clash-proxy-parser/public/rules/copymanga.yaml',
      path: './ruleset/copymanga.yaml',
      interval: 86400,
    },
  }
  config.rules = [
    'RULE-SET,applications,BitTorrent',
    'DOMAIN-KEYWORD,steamserver.net,DIRECT',
    'DOMAIN-KEYWORD,hf-mirror.com,DIRECT',
    'DOMAIN-KEYWORD,exhentai.org,e-hentai',
    'DOMAIN-KEYWORD,e-hentai.org,e-hentai',
    'DOMAIN,clash.razord.top,DIRECT',
    'DOMAIN,yacd.haishan.me,DIRECT',
    'RULE-SET,private,DIRECT',
    'RULE-SET,copymanga,PROXY',
    'RULE-SET,fanqie,REJECT',
    'RULE-SET,reject,REJECT',
    'RULE-SET,icloud,DIRECT',
    'RULE-SET,apple,DIRECT',
    'RULE-SET,google,DIRECT',
    'RULE-SET,proxy,PROXY',
    'RULE-SET,direct,DIRECT',
    'RULE-SET,lancidr,DIRECT',
    'RULE-SET,cncidr,DIRECT',
    'RULE-SET,telegramcidr,PROXY',
    'GEOIP,LAN,DIRECT',
    'GEOIP,CN,DIRECT',
    'MATCH,Final',
  ]

  const groupList = Object.entries({
    香港: ['香港', 'Hong Kong', 'HK'],
    台湾: ['台湾', 'Taiwan', 'TW'],
    美国: ['美国', 'United States', 'US'],
    日本: ['日本', 'Japan', 'JP'],
    新加坡: ['新加坡', 'Singapore', 'SG'],
    实验性: ['实验性'],
    标准: ['标准'],
    高级: ['高级'],
  })
    .map(([name, keys]) => {
      return {
        name,
        type: 'url-test',
        url: 'http://www.gstatic.com/generate_204',
        interval: 86400,
        proxies: config.proxies
          .filter((proxy) => keys.some((key) => proxy.name.includes(key)))
          .map((proxy) => proxy.name),
      }
    })
    .filter((o) => {
      return o.proxies.length > 0
    })

  const groupNameList = groupList.map((o) => o.name)

  config['proxy-groups'].push({
    name: 'Final',
    type: 'select',
    proxies: ['DIRECT', 'PROXY'],
  })
  config['proxy-groups'].push({
    name: 'PROXY',
    type: 'select',
    proxies: ['指定节点', '自动选择', '故障转移', ...groupNameList],
  })
  config['proxy-groups'].push({
    name: '指定节点',
    type: 'select',
    proxies: config.proxies.map((proxy) => proxy.name),
  })
  config['proxy-groups'].push({
    name: 'BitTorrent',
    type: 'select',
    proxies: ['DIRECT', 'PROXY'],
  })
  config['proxy-groups'].push({
    name: 'e-hentai',
    type: 'select',
    proxies: ['指定节点', '自动选择', '故障转移', ...config.proxies.map((proxy) => proxy.name)],
  })
  config['proxy-groups'].push({
    name: '自动选择',
    type: 'url-test',
    url: 'http://www.gstatic.com/generate_204',
    interval: 7200,
    proxies: config.proxies.map((proxy) => proxy.name),
  })
  config['proxy-groups'].push({
    name: '故障转移',
    type: 'fallback',
    url: 'http://www.gstatic.com/generate_204',
    interval: 7200,
    proxies: config.proxies.map((proxy) => proxy.name),
  })

  config['proxy-groups'] = config['proxy-groups'].concat(groupList)
  config['proxy-groups'] = config['proxy-groups'].filter((group) => group.proxies.length > 0)

  return config
}
