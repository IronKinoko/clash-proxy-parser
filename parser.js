module.exports.parse = (raw, { yaml, console }) => {
  const config = yaml.parse(raw)

  config['proxy-groups'] = []
  config['rule-providers'] = {
    reject: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/reject.txt',
      path: './ruleset/reject.yaml',
      interval: 86400,
    },
    icloud: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/icloud.txt',
      path: './ruleset/icloud.yaml',
      interval: 86400,
    },
    apple: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/apple.txt',
      path: './ruleset/apple.yaml',
      interval: 86400,
    },
    google: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/google.txt',
      path: './ruleset/google.yaml',
      interval: 86400,
    },
    proxy: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/proxy.txt',
      path: './ruleset/proxy.yaml',
      interval: 86400,
    },
    direct: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/direct.txt',
      path: './ruleset/direct.yaml',
      interval: 86400,
    },
    private: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/private.txt',
      path: './ruleset/private.yaml',
      interval: 86400,
    },
    gfw: {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/gfw.txt',
      path: './ruleset/gfw.yaml',
      interval: 86400,
    },
    'tld-not-cn': {
      type: 'http',
      behavior: 'domain',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/tld-not-cn.txt',
      path: './ruleset/tld-not-cn.yaml',
      interval: 86400,
    },
    telegramcidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/telegramcidr.txt',
      path: './ruleset/telegramcidr.yaml',
      interval: 86400,
    },
    cncidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/cncidr.txt',
      path: './ruleset/cncidr.yaml',
      interval: 86400,
    },
    lancidr: {
      type: 'http',
      behavior: 'ipcidr',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt',
      path: './ruleset/lancidr.yaml',
      interval: 86400,
    },
    applications: {
      type: 'http',
      behavior: 'classical',
      url: 'https://testingcf.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/applications.txt',
      path: './ruleset/applications.yaml',
      interval: 86400,
    },
  }
  config.rules = [
    'RULE-SET,applications,DIRECT',
    'DOMAIN,clash.razord.top,DIRECT',
    'DOMAIN,yacd.haishan.me,DIRECT',
    'RULE-SET,private,DIRECT',
    'RULE-SET,reject,REJECT',
    'RULE-SET,icloud,DIRECT',
    'RULE-SET,apple,DIRECT',
    'RULE-SET,google,DIRECT',
    'RULE-SET,proxy,PROXY',
    'RULE-SET,direct,DIRECT',
    'RULE-SET,lancidr,DIRECT',
    'RULE-SET,cncidr,DIRECT',
    'RULE-SET,telegramcidr,PROXY',
    'GEOIP,LAN,DIRECT',
    'GEOIP,CN,DIRECT',
    'MATCH,PROXY',
  ]

  const groupNameList = ['香港', '台湾', '美国', '日本']

  config['proxy-groups'].push({
    name: 'PROXY',
    type: 'select',
    proxies: ['指定节点', '自动选择', '故障转移', ...groupNameList],
  })

  config['proxy-groups'].push({
    name: '指定节点',
    type: 'select',
    proxies: config.proxies.map((proxy) => proxy.name),
  })
  config['proxy-groups'].push({
    name: '自动选择',
    type: 'url-test',
    url: 'http://www.gstatic.com/generate_204',
    interval: 7200,
    proxies: config.proxies.map((proxy) => proxy.name),
  })
  config['proxy-groups'].push({
    name: '故障转移',
    type: 'fallback',
    url: 'http://www.gstatic.com/generate_204',
    interval: 7200,
    proxies: config.proxies.map((proxy) => proxy.name),
  })

  // add proxy-groups by groupNameList
  groupNameList.forEach((name) => {
    config['proxy-groups'].push({
      name,
      type: 'url-test',
      url: 'http://www.gstatic.com/generate_204',
      interval: 86400,
      proxies: config.proxies
        .filter((proxy) => proxy.name.includes(name))
        .map((proxy) => proxy.name),
    })
  })

  config['proxy-groups'] = config['proxy-groups'].filter((group) => group.proxies.length > 0)

  return yaml.stringify(config)
}
