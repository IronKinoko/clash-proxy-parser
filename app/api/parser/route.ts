import { NextRequest } from 'next/server'
import { parser } from './parser'
import { pick } from 'lodash-es'

export async function GET(req: NextRequest) {
  const url = req.nextUrl.searchParams.get('url')

  if (!url) {
    return new Response('URL parameter is required', { status: 400 })
  }

  const res = await fetch(url, { headers: { 'user-agent': 'ClashforWindows/0.19.23' } })
  const raw = await res.text()
  const ret = parser(raw)

  return new Response(ret, {
    headers: pick(Object.fromEntries(res.headers.entries()), [
      'content-disposition',
      'subscription-userinfo',
    ]),
  })
}
