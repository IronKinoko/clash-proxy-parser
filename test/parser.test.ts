import parser from '../app/api/parser/parser.js'
import yaml from 'yaml'
import fs from 'fs'
import path from 'path'

// 使用 fixtures 文件夹
const raw = fs.readFileSync(path.join(__dirname, 'fixtures', 'demo.yaml'), 'utf-8')

describe('parser', () => {
  it('should parse the config correctly', () => {
    const config = parser(yaml.parse(raw))
    expect(config).toBeDefined()
    expect(yaml.stringify(config)).toMatchSnapshot()
  })
})
