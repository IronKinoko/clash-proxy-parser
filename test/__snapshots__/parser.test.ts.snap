// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`parser should parse the config correctly 1`] = `
"proxies:
  - name: 🇭🇰 Hong Kong 01
  - name: 🇭🇰 Hong Kong 02
  - name: 🇭🇰 Hong Kong 03
  - name: 🇭🇰 Hong Kong 04
  - name: 🇭🇰 Hong Kong 05
  - name: 🇹🇼 Taiwan 01
  - name: 🇹🇼 Taiwan 02
  - name: 🇹🇼 Taiwan 03
  - name: 🇹🇼 Taiwan 04
  - name: 🇹🇼 Taiwan 05
  - name: 🇸🇬 Singapore 01
  - name: 🇸🇬 Singapore 02
  - name: 🇸🇬 Singapore 03
  - name: 🇸🇬 Singapore 04
  - name: 🇸🇬 Singapore 05
  - name: 🇯🇵 Japan 01
  - name: 🇯🇵 Japan 02
  - name: 🇯🇵 Japan 03
  - name: 🇯🇵 Japan 04
  - name: 🇯🇵 Japan 05
  - name: 🇺🇸 United States 01
  - name: 🇺🇸 United States 02
  - name: 🇺🇸 United States 03
  - name: 🇺🇸 United States 04
  - name: 🇺🇸 United States 05
  - name: 🇩🇪 Germany 01
  - name: 🇬🇧 Britain 01
  - name: 🇬🇧 Britain 02
  - name: 🇦🇷 Argentina 01
  - name: 🇹🇷 Turkey 01
  - name: 🇮🇳 India 01
proxy-groups:
  - name: 漏网之鱼
    type: select
    proxies:
      - DIRECT
      - PROXY
  - name: PROXY
    type: select
    proxies:
      - 指定节点
      - 自动选择
      - 故障转移
      - 🇭🇰 香港
      - 🇭🇰 香港 均衡
      - 🇹🇼 台湾
      - 🇹🇼 台湾 均衡
      - 🇺🇸 美国
      - 🇺🇸 美国 均衡
      - 🇯🇵 日本
      - 🇯🇵 日本 均衡
      - 🇸🇬 新加坡
      - 🇸🇬 新加坡 均衡
      - 其他
      - 其他 均衡
  - name: 指定节点
    type: select
    proxies: &a1
      - 🇭🇰 Hong Kong 01
      - 🇭🇰 Hong Kong 02
      - 🇭🇰 Hong Kong 03
      - 🇭🇰 Hong Kong 04
      - 🇭🇰 Hong Kong 05
      - 🇹🇼 Taiwan 01
      - 🇹🇼 Taiwan 02
      - 🇹🇼 Taiwan 03
      - 🇹🇼 Taiwan 04
      - 🇹🇼 Taiwan 05
      - 🇸🇬 Singapore 01
      - 🇸🇬 Singapore 02
      - 🇸🇬 Singapore 03
      - 🇸🇬 Singapore 04
      - 🇸🇬 Singapore 05
      - 🇯🇵 Japan 01
      - 🇯🇵 Japan 02
      - 🇯🇵 Japan 03
      - 🇯🇵 Japan 04
      - 🇯🇵 Japan 05
      - 🇺🇸 United States 01
      - 🇺🇸 United States 02
      - 🇺🇸 United States 03
      - 🇺🇸 United States 04
      - 🇺🇸 United States 05
      - 🇩🇪 Germany 01
      - 🇬🇧 Britain 01
      - 🇬🇧 Britain 02
      - 🇦🇷 Argentina 01
      - 🇹🇷 Turkey 01
      - 🇮🇳 India 01
  - name: 自动选择
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
    proxies: *a1
  - name: 故障转移
    type: fallback
    url: http://www.gstatic.com/generate_204
    interval: 60
    proxies: *a1
  - name: BitTorrent
    type: select
    proxies:
      - DIRECT
      - PROXY
  - name: e-hentai
    type: select
    proxies:
      - 指定节点
      - 自动选择
      - 故障转移
      - 🇭🇰 香港
      - 🇭🇰 香港 均衡
      - 🇹🇼 台湾
      - 🇹🇼 台湾 均衡
      - 🇺🇸 美国
      - 🇺🇸 美国 均衡
      - 🇯🇵 日本
      - 🇯🇵 日本 均衡
      - 🇸🇬 新加坡
      - 🇸🇬 新加坡 均衡
      - 其他
      - 其他 均衡
      - 🇭🇰 Hong Kong 01
      - 🇭🇰 Hong Kong 02
      - 🇭🇰 Hong Kong 03
      - 🇭🇰 Hong Kong 04
      - 🇭🇰 Hong Kong 05
      - 🇹🇼 Taiwan 01
      - 🇹🇼 Taiwan 02
      - 🇹🇼 Taiwan 03
      - 🇹🇼 Taiwan 04
      - 🇹🇼 Taiwan 05
      - 🇸🇬 Singapore 01
      - 🇸🇬 Singapore 02
      - 🇸🇬 Singapore 03
      - 🇸🇬 Singapore 04
      - 🇸🇬 Singapore 05
      - 🇯🇵 Japan 01
      - 🇯🇵 Japan 02
      - 🇯🇵 Japan 03
      - 🇯🇵 Japan 04
      - 🇯🇵 Japan 05
      - 🇺🇸 United States 01
      - 🇺🇸 United States 02
      - 🇺🇸 United States 03
      - 🇺🇸 United States 04
      - 🇺🇸 United States 05
      - 🇩🇪 Germany 01
      - 🇬🇧 Britain 01
      - 🇬🇧 Britain 02
      - 🇦🇷 Argentina 01
      - 🇹🇷 Turkey 01
      - 🇮🇳 India 01
  - name: 🇭🇰 香港
    proxies: &a2
      - 🇭🇰 Hong Kong 01
      - 🇭🇰 Hong Kong 02
      - 🇭🇰 Hong Kong 03
      - 🇭🇰 Hong Kong 04
      - 🇭🇰 Hong Kong 05
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 🇭🇰 香港 均衡
    proxies: *a2
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
  - name: 🇹🇼 台湾
    proxies: &a3
      - 🇹🇼 Taiwan 01
      - 🇹🇼 Taiwan 02
      - 🇹🇼 Taiwan 03
      - 🇹🇼 Taiwan 04
      - 🇹🇼 Taiwan 05
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 🇹🇼 台湾 均衡
    proxies: *a3
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
  - name: 🇺🇸 美国
    proxies: &a4
      - 🇺🇸 United States 01
      - 🇺🇸 United States 02
      - 🇺🇸 United States 03
      - 🇺🇸 United States 04
      - 🇺🇸 United States 05
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 🇺🇸 美国 均衡
    proxies: *a4
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
  - name: 🇯🇵 日本
    proxies: &a5
      - 🇯🇵 Japan 01
      - 🇯🇵 Japan 02
      - 🇯🇵 Japan 03
      - 🇯🇵 Japan 04
      - 🇯🇵 Japan 05
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 🇯🇵 日本 均衡
    proxies: *a5
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
  - name: 🇸🇬 新加坡
    proxies: &a6
      - 🇸🇬 Singapore 01
      - 🇸🇬 Singapore 02
      - 🇸🇬 Singapore 03
      - 🇸🇬 Singapore 04
      - 🇸🇬 Singapore 05
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 🇸🇬 新加坡 均衡
    proxies: *a6
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
  - name: 其他
    proxies: &a7
      - 🇩🇪 Germany 01
      - 🇬🇧 Britain 01
      - 🇬🇧 Britain 02
      - 🇦🇷 Argentina 01
      - 🇹🇷 Turkey 01
      - 🇮🇳 India 01
    type: url-test
    url: http://www.gstatic.com/generate_204
    interval: 180
  - name: 其他 均衡
    proxies: *a7
    type: load-balance
    strategy: consistent-hashing
    url: http://www.gstatic.com/generate_204
    interval: 600
rule-providers:
  reject:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/reject.txt
    path: ./ruleset/reject.yaml
    interval: 86400
  icloud:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/icloud.txt
    path: ./ruleset/icloud.yaml
    interval: 86400
  apple:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/apple.txt
    path: ./ruleset/apple.yaml
    interval: 86400
  google:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/google.txt
    path: ./ruleset/google.yaml
    interval: 86400
  proxy:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/proxy.txt
    path: ./ruleset/proxy.yaml
    interval: 86400
  direct:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/direct.txt
    path: ./ruleset/direct.yaml
    interval: 86400
  private:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/private.txt
    path: ./ruleset/private.yaml
    interval: 86400
  gfw:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/gfw.txt
    path: ./ruleset/gfw.yaml
    interval: 86400
  tld-not-cn:
    type: http
    behavior: domain
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/tld-not-cn.txt
    path: ./ruleset/tld-not-cn.yaml
    interval: 86400
  telegramcidr:
    type: http
    behavior: ipcidr
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/telegramcidr.txt
    path: ./ruleset/telegramcidr.yaml
    interval: 86400
  cncidr:
    type: http
    behavior: ipcidr
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/cncidr.txt
    path: ./ruleset/cncidr.yaml
    interval: 86400
  lancidr:
    type: http
    behavior: ipcidr
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt
    path: ./ruleset/lancidr.yaml
    interval: 86400
  applications:
    type: http
    behavior: classical
    url: https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/applications.txt
    path: ./ruleset/applications.yaml
    interval: 86400
  fanqie:
    type: http
    behavior: classical
    url: https://cdn.jsdelivr.net/gh/ironkinoko/clash-proxy-parser/public/rules/fanqie.yaml
    path: ./ruleset/fanqie.yaml
    interval: 86400
  copymanga:
    type: http
    behavior: classical
    url: https://cdn.jsdelivr.net/gh/ironkinoko/clash-proxy-parser/public/rules/copymanga.yaml
    path: ./ruleset/copymanga.yaml
    interval: 86400
rules:
  - RULE-SET,applications,BitTorrent
  - DOMAIN-KEYWORD,steamserver.net,DIRECT
  - DOMAIN-KEYWORD,hf-mirror.com,DIRECT
  - DOMAIN-KEYWORD,exhentai.org,e-hentai
  - DOMAIN-KEYWORD,e-hentai.org,e-hentai
  - DOMAIN,clash.razord.top,DIRECT
  - DOMAIN,yacd.haishan.me,DIRECT
  - RULE-SET,private,DIRECT
  - RULE-SET,copymanga,PROXY
  - RULE-SET,fanqie,REJECT
  - RULE-SET,reject,REJECT
  - RULE-SET,icloud,DIRECT
  - RULE-SET,apple,DIRECT
  - RULE-SET,google,DIRECT
  - RULE-SET,proxy,PROXY
  - RULE-SET,direct,DIRECT
  - RULE-SET,lancidr,DIRECT
  - RULE-SET,cncidr,DIRECT
  - RULE-SET,telegramcidr,PROXY
  - GEOIP,LAN,DIRECT
  - GEOIP,CN,DIRECT
  - MATCH,漏网之鱼
"
`;
